libs/stocks/feature-shell/src/lib/stocks/stocks.component.html

```diff
@@ -16,18 +16,72 @@
       >
     </mat-error>
   </mat-form-field>
+  <mat-form-field>
+    <mat-label>From date:</mat-label>
+    <input
+      [class.error-field]="shouldDisplayDateError"
+      matInput
+      formControlName="from"
+      [min]="minDate"
+      [max]="maxDate"
+      [matDatepicker]="from"
+      placeholder="Choose a date"
+    />
+    <mat-error>
+      <span
+        *ngIf="
+          !stockPickerForm.get('from').valid &&
+          stockPickerForm.get('from').touched &&
+          !stockPickerForm.get('from').value
+        "
+        >From date is required</span
+      >
+      <mat-error
+        *ngIf="stockPickerForm.get('from').hasError('matDatepickerMin')"
+        >Date cannot be more than 15 years ago</mat-error
+      >
+      <mat-error
+        *ngIf="stockPickerForm.get('from').hasError('matDatepickerMax')"
+        >Date cannot be in future</mat-error
+      >
+    </mat-error>
+    <mat-datepicker-toggle matSuffix [for]="from"></mat-datepicker-toggle>
+    <mat-datepicker #from></mat-datepicker>
+  </mat-form-field>
 
   <mat-form-field>
-    <mat-label>Favorite time period</mat-label>
-    <mat-select formControlName="period">
-      <mat-option
-        *ngFor="let timePeriod of timePeriods"
-        [value]="timePeriod.value"
+    <mat-label>To date:</mat-label>
+    <input
+      [class.error-field]="shouldDisplayDateError"
+      matInput
+      formControlName="to"
+      [min]="minDate"
+      [max]="maxDate"
+      [matDatepicker]="to"
+      placeholder="Choose a date"
+    />
+    <mat-error>
+      <span
+        *ngIf="
+          !stockPickerForm.get('to').valid &&
```

Comment:
This complicated of a conditional in the HTML could go in a getter in the component